src/journal/journald-context.c

```diff
@@ -202,6 +202,17 @@ static ClientContext* client_context_free(Server *s, ClientContext *c) {
         return mfree(c);
 }
 
+static void client_context_drop_from_cache(Server *s, ClientContext *c) {
+        assert(s);
+
+        assert(c->in_lru);
```

Comment:
hmm, you are calling this for pinned entries too afaics, so this assert should hit all the time. what am i missing?